src/base-element.js

```diff
@@ -369,6 +369,25 @@ export class BaseElement {
   }
 
   /**
+   * Subclasses can override this method to create an analytics config.
+   * @returns {?Promise<!JSONType>}
```

Comment:
Why is the other way better?  I like `{?Promise<!JSONType>}` because it saves one promise call and returns a bit earlier